spring-integration-file/src/main/java/org/springframework/integration/file/filters/FileNameFileListFilter.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.file.filters;
+
+/**
+ * A marker interface for {@link FileListFilter}s that indicates the filter
+ * examines file names.
+ *
+ * @param <F> the type of the object being filtered.
+ *
+ * @author Gary Russell
+ * @since 5.0
+ *
+ */
+public interface FileNameFileListFilter<F> extends FileListFilter<F> {
```

Comment:
Gary, any strong requirements to have this extension?
Looks like it doesn't affect `AbstractMarkerFilePresentFileListFilter` unless there might be some stranger to inject an `AbstractMarkerFilePresentFileListFilter` into the `AbstractMarkerFilePresentFileListFilter`.
Thanks